### PR TITLE
TLT-3185 Blueprint Courses

### PR DESCRIPTION
**Pull request by Chris-Thornton-Harvard**
[TLT-3185](https://jira.huit.harvard.edu/browse/TLT-3185)
This adds the newly released feature of creating a Canvas blueprint course.
Accounts selected to be a blueprint course in the 'Canvas Site Creator' will have the BLU- prefix and be created at the school level rather than the ILE or SB sub-accounts.

This is currently waiting on the 0.10.2 release of the Canvas Python SDK located at
https://github.com/penzance/canvas_python_sdk/releases
Until that is released, the update functionality will not work on dev and you will receive an error message.

### CODE REVIEW

File: canvas_site_creator/api.py [hunk 1]
```diff
@@ -236,8 +238,19 @@ def create_canvas_course_and_section(request):
...
+        if is_blueprint:
+            update_parameters = dict(
+                request_ctx=SDK_CONTEXT,
+                id=course_result['id'],
+                course_blueprint=True
+            )
+            update_result = update_course(**update_parameters).json()
     except Exception as e:
         message = u'Error creating new course via SDK with request={}'.format(
             request_parameters)
```
Comment (sapnamysore) [inline canvas_site_creator/api.py]: The exception block is for both the canvas calls.However the logging is relevant to the create call failing and prints the request_parameters even if there would be an exception with the second call. Maybe you could move the  update_course call into a separate block and log the update_parameters in the error message?

File: canvas_site_creator/static/canvas_site_creator/js/controllers/CreateNewCourseController.js [hunk 2]
```diff
@@ -224,30 +237,37 @@
...
             var data = {
-                departments: [departmentId],
                 registrar_code: $scope.course.code,
                 registrar_code_display: $scope.course.code,
-                school: $scope.school.id,
+                school: $scope.school.id
             };
+            // Only include departments if this is not a blueprint course
+            if(!$scope.isBlueprint) {
+                data.departments = [departmentId]
```
Comment (sapnamysore) [inline canvas_site_creator/static/canvas_site_creator/js/controllers/CreateNewCourseController.js]: missing semicolon at the end of the line?

Comment (Chris-Thornton-Harvard) [reply canvas_site_creator/static/canvas_site_creator/js/controllers/CreateNewCourseController.js]: Good catch!

File: canvas_site_creator/api.py (hunk 1 above)
Comment (Chris-Thornton-Harvard) [reply canvas_site_creator/api.py]: I added a nested try except to log the update parameters and exception details if there is an error during the course update.

**Pull request merged**